internal/datastore/common/sql.go

```diff
@@ -325,12 +351,18 @@ func (sqf SchemaQueryFilterer) MustFilterWithRelationshipsFilter(filter datastor
 }
 
 func (sqf SchemaQueryFilterer) FilterWithRelationshipsFilter(filter datastore.RelationshipsFilter) (SchemaQueryFilterer, error) {
-	sqf = sqf.FilterToResourceType(filter.ResourceType)
+	if filter.OptionalResourceType != "" {
+		sqf = sqf.FilterToResourceType(filter.OptionalResourceType)
+	}
 
 	if filter.OptionalResourceRelation != "" {
 		sqf = sqf.FilterToRelation(filter.OptionalResourceRelation)
 	}
 
+	if len(filter.OptionalResourceIds) > 0 && filter.OptionalResourceIDPrefix != "" {
```

Comment:
This does not have a unit test, perhaps add one to assert that the condition will panic? (even though this won't happen as it is already checked in the service controllers)

Reply:
I don't generally like to assert panics and this is only a "break glass in case of emergency" panic (for the reason you specified), so I'm not 100% convinced it needs a unit test